pkg/manager/member/tidb_member_manager.go

```diff
@@ -208,7 +208,7 @@ func (tmm *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.Tid
 	}
 
 	if tmm.autoFailover {
-		if tc.TiDBAllPodsStarted() && tc.TiDBAllMembersReady() && tc.Status.TiDB.FailureMembers != nil {
+		if (tc.TiDBAllPodsStarted() && tc.TiDBAllMembersReady() && tc.Status.TiDB.FailureMembers != nil) || *newTiDBSet.Spec.Replicas == int32(0) {
```

Comment:
I think we should not perform `Recover` when `.failureMembers` is empty

Reply:
agree